File: node/src/safeField.js
'use strict';

// L-8: strip/escape control characters from untrusted values before they become log fields.

const CONTROL_CHARS = /[\r\n\t\x00-\x1f\x7f]/g;

function safeField(value) {
  if (typeof value !== 'string') return value;
  return value.replace(CONTROL_CHARS, ' ');
}

module.exports = { safeField };

File: python/zap_logging/safe_field.py
"""L-8: strip/escape control characters from untrusted values before they become log fields."""

import re

_CONTROL_CHARS = re.compile(r"[\r\n\t\x00-\x1f\x7f]")


def safe_field(value):
    if not isinstance(value, str):
        return value
    return _CONTROL_CHARS.sub(" ", value)
